File: Back-end/FITExamAPI/FITExamAPI/Models/Exam.cs
﻿using System.ComponentModel.DataAnnotations;

namespace FITExamAPI.Models
{
    public class Exam
    {
        public int Id { get; set; }
        [MaxLength(255)]
        public string? Name { get; set; } = string.Empty;
        public string? SubjectId { get; set; }
        public int? CreatorId { get; set; }
        public int? ExamTime { get; set; }
        public int? NumberOfQuestions { get; set; }
        public DateTime? StartDate { get; set; }
        public DateTime? EndDate { get; set; }
        public virtual Subject? Subject { get; set; }
        public virtual User? User { get; set; }
        public virtual ICollection<Question>? Questions { get; set; } = new List<Question>();
        public virtual ICollection<Result>? Results { get; set; } = new List<Result>();

    }
}

File: Front-end/FIT_Exam/app/src/main/java/vn/edu/hcmuaf/fit/fitexam/model/Exam.java
package vn.edu.hcmuaf.fit.fitexam.model;

import java.io.Serializable;
import java.util.List;

public class Exam implements Serializable {
    private int id;
    private String name;
    private int subjectId;
    private int creatorId;
    private int examTime;
    private int numberOfQuestions;
    private int minimumDifficulty;
    private int maximumDifficulty;
    private Subject subject;
    private User user;
    private List<Result> results;

    public Exam() {
    }

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public int getSubjectId() {
        return subjectId;
    }

    public void setSubjectId(int subjectId) {
        this.subjectId = subjectId;
    }

    public int getCreatorId() {
        return creatorId;
    }

    public void setCreatorId(int creatorId) {
        this.creatorId = creatorId;
    }

    public int getExamTime() {
        return examTime;
    }

    public void setExamTime(int examTime) {
        this.examTime = examTime;
    }

    public int getNumberOfQuestions() {
        return numberOfQuestions;
    }

    public void setNumberOfQuestions(int numberOfQuestions) {
        this.numberOfQuestions = numberOfQuestions;
    }

    public int getMinimumDifficulty() {
        return minimumDifficulty;
    }

    public void setMinimumDifficulty(int minimumDifficulty) {
        this.minimumDifficulty = minimumDifficulty;
    }

    public int getMaximumDifficulty() {
        return maximumDifficulty;
    }

    public void setMaximumDifficulty(int maximumDifficulty) {
        this.maximumDifficulty = maximumDifficulty;
    }

    public Subject getSubject() {
        return subject;
    }

    public void setSubject(Subject subject) {
        this.subject = subject;
    }

    public User getUser() {
        return user;
    }

    public void setUser(User user) {
        this.user = user;
    }

    public List<Result> getResults() {
        return results;
    }

    public void setResults(List<Result> results) {
        this.results = results;
    }

}
